llvm/test/CodeGen/PowerPC/pr59074.ll

```diff
@@ -0,0 +1,132 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+; RUN: llc -mtriple=powerpc64le-unknown-linux-gnu -mcpu=pwr7 < %s | FileCheck %s --check-prefix=LE64
+; RUN: llc -mtriple=powerpcle-unknown-linux-gnu -mcpu=pwr7 < %s | FileCheck %s --check-prefix=LE32
+; RUN: llc -mtriple=powerpc64-ibm-aix -mcpu=pwr7 < %s | FileCheck %s --check-prefix=BE64
+; RUN: llc -mtriple=powerpc-ibm-aix -mcpu=pwr7 < %s | FileCheck %s --check-prefix=BE32
```

Comment:
`-mcpu=pwr8` touches another codepath (such instructions are not generated, so only crash on `-mcpu=pwr7`).

The crash comes from constant shift amounts of `srl_parts`, which only presents after splitting `v2i128` into `v1i128` (from the subtraction parts).